openstudiocore/src/openstudio_lib/OSDropZone.cpp

```diff
@@ -679,10 +682,31 @@ void OSDropZone2::dropEvent(QDropEvent *event)
 {
   event->accept();
 
+  std::shared_ptr<OSDocument> doc = OSAppBase::instance()->currentDocument();
+
   if((event->proposedAction() == Qt::CopyAction) && m_modelObject && m_set){
 
     OSItemId itemId(event->mimeData());
-    boost::optional<model::ModelObject> modelObject = OSAppBase::instance()->currentDocument()->getModelObject(itemId);
+
+    boost::optional<model::ModelObject> modelObject;
+
+    // If what you dragged is from the BCL, then VT it and insert it in model
+    // TODO: should we modify OSDocument::getModelObject instead?
+    if(itemId.sourceId() == OSItemId::BCL_SOURCE_ID)
```

Comment:
@jmarrec I see now, yes please update OSDocument::getModelObject instead

Reply:
Hum, now that I look into it, `OSDocument::getModelObject` is actually marked `const`. So if I modify it to insert the component into model if fromBCL, I have to remove the `const`. What do you think @macumber ?